lib/edgehog_device/include/edgehog_private.h

```diff
@@ -19,32 +19,61 @@
 
 #include <astarte_device_sdk/device.h>
 
+/** @brief Possible states for the Edgehog device. */
+enum device_states
+{
+    /** @brief The device is not operational. */
+    DEVICE_STOPPED = 0U,
+    /** @brief The device is has been started, but does not yet have connectivity. */
+    DEVICE_STARTING,
+    /** @brief The device is has been started, and has been connected to Astarte. */
+    DEVICE_CONNECTED,
+    /** @brief The device is fully operational. */
+    DEVICE_RUNNING,
+};
+
 /**
  * @brief Internal struct for an instance of an Edgehog device.
  *
- * @warning Users should not modify the content of this struct directly
+ * @warning Users should not modify the content of this struct directly.
  */
 struct edgehog_device_t
 {
+    /** @brief Edgehog device state. */
+    enum device_states state;
     /** @brief Handle of an Astarte device. */
     astarte_device_handle_t astarte_device;
+    /** @brief Original connection callback provided by the used, might be NULL. */
+    astarte_device_connection_cbk_t original_connection_cbk;
```

Comment:
What do you think to rename it to `user`?
```suggestion
    astarte_device_connection_cbk_t user_connection_cbk;
```